src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -530,12 +530,17 @@ private async void ReconnectLoop()
             await DisposeSocketAsync(true).ConfigureAwait(false);
 
             var defaultScheme = _currentConnectUri!.Uri.Scheme;
-            var urls = (Opts.NoRandomize
-                           ? WritableServerInfo?.ClientConnectUrls?.Select(x => new NatsUri(x, false, defaultScheme)).Distinct().ToArray()
-                           : WritableServerInfo?.ClientConnectUrls?.Select(x => new NatsUri(x, false, defaultScheme)).OrderBy(_ => Guid.NewGuid()).Distinct().ToArray())
-                       ?? Array.Empty<NatsUri>();
-            if (urls.Length == 0)
-                urls = Opts.GetSeedUris();
+            var serverReportedUrls = ServerInfo?
+                                         .ClientConnectUrls?
+                                         .Select(x => new NatsUri(x, false, defaultScheme))
+                                     ?? Array.Empty<NatsUri>();
+
+            // Always keep the original seed URLs in the list of URLs to connect to
+            var connectUrls = serverReportedUrls.Concat(Opts.GetSeedUris());
+
+            var urls = Opts.NoRandomize
+                ? connectUrls.Distinct().ToArray()
+                : connectUrls.OrderBy(_ => Guid.NewGuid()).Distinct().ToArray();
 
             // add last.
```

Comment:
Nit but is it a little cleaner if we drop the ternary?

```suggestion
            var urls = serverReportedUrls.Concat(Opts.GetSeedUris()).Distinct();
            if (!Opts.NoRandomize)
            {
                urls = urls.OrderBy(_ => Guid.NewGuid());
            }

            // Ensure the current URL is last in the list
```

Reply:
nice!